homeassistant/components/homematicip.py

```diff
@@ -0,0 +1,103 @@
+"""
+Support for HomematicIP components.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/homematicip/
+"""
+
+import logging
+from datetime import timedelta
+from socket import timeout
+
+import voluptuous as vol
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+
+REQUIREMENTS = ['homematicip==0.8']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'homematicip'
+
+CONF_NAME = 'name'
+CONF_ACCESSPOINT = 'accesspoint'
+CONF_AUTHTOKEN = 'authtoken'
+CONF_TIMEOUT = 'timeout'
+CONF_RECONNECT = 'reconnect'
+CONF_DISCOVERY = 'discovery'
+
+CONFIG_SCHEMA = vol.Schema({
+    vol.Optional(DOMAIN): [vol.Schema({
+        vol.Optional(CONF_NAME, default=''): cv.string,
+        vol.Required(CONF_ACCESSPOINT): cv.string,
+        vol.Required(CONF_AUTHTOKEN): cv.string,
+        vol.Optional(CONF_TIMEOUT, default=timedelta(seconds=20)): (
```

Comment:
I don't think that it makes sense to allow users to pick a timeout. You should pick one and that's it.

Reply:
Actually, you're not using it so just remove it.